### PR TITLE
progress on simulation/test target

### PR DESCRIPTION
**Pull request by codingUniv**

### CODE REVIEW

File: hardware/simulation/icarus.mk [hunk 1]
```diff
@@ -4,13 +4,15 @@ ifeq ($(VCD),1)
...
+TEST_LOG=test.log
+
 comp: a.out
 
 a.out: $(VHDR) $(VSRC)
 	iverilog $(VFLAGS)
 
 exec:
-	./a.out $(TEST_LOG)
+	./a.out | tee -a $(TEST_LOG)
```
Comment (jjts) [inline hardware/simulation/icarus.mk:15]: do not reate a variable if you are using it only once. Just use test.log

File: hardware/simulation/simulation.mk [hunk 2]
```diff
@@ -69,7 +69,11 @@ ifneq ($(SIM_SERVER),)
 	ssh $(SIM_SSH_FLAGS) $(SIM_USER)@$(SIM_SERVER) 'if [ -f $(REMOTE_BUILD_DIR)/fpga/Makefile ]; then make -C $(REMOTE_BUILD_DIR)/sim clean; fi'
 endif
 
-test: clean $(TEST_LIST) 
+clean_t:
+	@rm -rf a.out test.log obj_dir
```
Comment (jjts) [inline hardware/simulation/simulation.mk:73]: this is icarus specific

File: hardware/simulation/simulation.mk [hunk 3]
```diff
@@ -69,7 +69,11 @@ ifneq ($(SIM_SERVER),)
...
+
+test: clean_t $(TEST_LIST) 
+	diff test.expected test.log
```
Comment (jjts) [inline hardware/simulation/simulation.mk:76]: this does not belong here

Comment (jjts) [reply hardware/simulation/simulation.mk:76]: belongs to the shared one in lib

File: hardware/simulation/verilator.mk [hunk 4]
```diff
@@ -7,12 +7,14 @@ ifeq ($(VCD),1)
...
+TEST_LOG=test.log
+
 comp: $(VHDR) $(VSRC)
 	echo $(TOP_MODULE)
 	verilator $(VFLAGS) $(WAVE)	
 	cd ./obj_dir && make -f V$(VTOP).mk
 
 exec:
-	./obj_dir/V$(VTOP) $(TEST_LOG)
+	./obj_dir/V$(VTOP) | tee -a $(TEST_LOG)
```
Comment (jjts) [inline hardware/simulation/verilator.mk:18]: remove test.log

**Comment by jjts** [comment]
This PR os on makefiles which duplicates my work and is not related to simulation

File: hardware/simulation/icarus.mk (hunk 1 above)
Comment (codingUniv) [reply hardware/simulation/icarus.mk:15]: OK

File: hardware/simulation/verilator.mk (hunk 4 above)
Comment (codingUniv) [reply hardware/simulation/verilator.mk:18]: OK

**Pull request merged**